File: submissions/1768-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        len1, len2 = len(word1), len(word2)
        i = 0
        res = []
        while (i < len1 or i < len2):
            if(i < len1):
                res.append(word1[i])
            if(i < len2):
                res.append(word2[i])
            i += 1
        
        return "".join(res)

File: submissions/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res;
        while(word1.size() && word2.size()) {
            res += word1[0];
            res += word2[0];
            word1 = word1.substr(1);
            word2 = word2.substr(1);
        }
        if(word1.size()) res += word1;
        if(word2.size()) res += word2;
        return res;
    }
};
